### DISCUSSION TITLE
cordova-plugin-advanced-http doesn't work on Android

### DISCUSSION THREAD

**Discussion by mraible** [ionic-team/capacitor#3622] [Q&A] [answered]
I created an [issue for this in the cordova-plugin-advanced-http](https://github.com/silkimen/cordova-plugin-advanced-http/issues/371) project, but I'm not sure it's caused by that plugin.

To summarize:

I'm getting an error when I try to run my Capacitor project in Android Studio:

```
/Users/mraible/dev/okta/schematics/apps/ionic-capacitor/android/capacitor-cordova-android-plugins/src/main/java/org/chromium/customtabsclient/shared/ServiceConnection.java:18: error: cannot find symbol
import android.support.customtabs.CustomTabsClient;
                                 ^
  symbol:   class CustomTabsClient
  location: package android.support.customtabs
```

I googled around a bit and landed on Android's [Custom Tabs Support Library](https://developer.android.com/topic/libraries/support-library/packages#custom-tabs). I figured I might need to add it, so I opened up `android/app/build.gradle`. In there, I see:

```groovy
apply from: 'capacitor.build.gradle'
```

And if I go to `capacitor.build.gradle`, it seems like the dependency is listed there:

```groovy
apply from: "../capacitor-cordova-android-plugins/cordova.variables.gradle"
dependencies {

    implementation "com.android.support:customtabs:23.2.0"
}
```

Any idea what I need to do to get things to compile?

**Comment by jcesarmobile** [comment, marked as answer]
The problem is the safari view controller plugin, not the http plugin.
Capacitor uses AndroidX dependencies, but that plugin uses an older non AndroidX dependency and that’s not possible on Android.

You can use jetifier to patch usages of old support libraries, see https://capacitorjs.com/docs/android/troubleshooting#error-package-android-support-does-not-exist

**Reply by mraible** [reply]
That worked - thank you so much! ❤️
